commons-numbers-complex-streams/src/main/java/org/apache/commons/numbers/complex/streams/ComplexUtils.java

```diff
@@ -979,31 +1118,30 @@ public static Complex extractComplexFromInterleavedArray(float[] f, int index) {
      * Converts a 2D interleaved complex {@code double[][]} array to a
      * {@code Complex[][]} array.
      *
-     * @param d 2D complex interleaved array
+     * @param i 2D complex interleaved array
      * @param interleavedDim Depth level of the array to interleave
      * @return 2D {@code Complex} array
-     *
-     * @since 4.0
+     * @throws IllegalArgumentException if {@code interleavedDim} is not 0 or 1
      */
-    public static Complex[][] interleaved2Complex(double[][] d, int interleavedDim) {
+    public static Complex[][] interleaved2Complex(double[][] i, int interleavedDim) {
         if (interleavedDim > 1 || interleavedDim < 0) {
-            new IndexOutOfRangeException(interleavedDim);
+            throw new IndexOutOfRangeException(interleavedDim);
         }
-        final int width = d.length;
-        final int height = d[0].length;
+        final int w = i.length;
+        final int h = i[0].length;
         Complex[][] c;
         if (interleavedDim == 0) {
-            c = new Complex[width / 2][height];
-            for (int x = 0; x < width / 2; x++) {
-                for (int y = 0; y < height; y++) {
-                    c[x][y] = new Complex(d[x * 2][y], d[x * 2 + 1][y]);
+            c = new Complex[w / 2][h];
+            for (int x = 0; x < w / 2; x++) {
+                for (int y = 0; y < h; y++) {
+                    c[x][y] = Complex.ofCartesian(i[x * 2][y], i[x * 2 + 1][y]);
                 }
             }
         } else {
-            c = new Complex[width][height / 2];
-            for (int x = 0; x < width; x++) {
-                for (int y = 0; y < height / 2; y++) {
-                    c[x][y] = new Complex(d[x][y * 2], d[x][y * 2 + 1]);
+            c = new Complex[w][h / 2];
+            for (int x = 0; x < w; x++) {
+                for (int y = 0; y < h / 2; y++) {
+                    c[x][y] = Complex.ofCartesian(i[x][y * 2], i[x][y * 2 + 1]);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).